standalone/detectors/fasttrack/include/PrefixTreeDepot.h

```diff
@@ -0,0 +1,278 @@
+
+#ifndef TREEDEPOT_HEADER_H
+#define TREEDEPOT_HEADER_H 1
+#pragma once
+
+/*
+ * DRace, a dynamic data race detector
+ *
+ * Copyright 2020 Siemens AG
+ *
+ * Authors:
+ *   Mihai Robescu <mihai-gabriel.robescu@siemens.com>
+ *   Felix Moessbauer <felix.moessbauer@siemens.com>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <deque>
+#include <memory>
+#include "PoolAllocator.h"
+
+/**
+ *------------------------------------------------------------------------------
+ *
+ * Header File that implements a prefix tree data structure to
+ * store call stack elements. This is one is optimized to be
+ * cache efficient, making each node a multiple of the size of the
+ * cache line
+ *
+ *------------------------------------------------------------------------------
+ */
+
+class INode {
```

Comment:
Hi, 
It is done like this, because when a node is created, it is first created through the abstract class and then specialized on the needs, hence I couldn't do with an abstract class and I had to find a way to not allow any client use the base function.

Reply:
@fmoessbauer